test/payments_refunds_previous_test.go

```diff
@@ -43,14 +59,37 @@ func TestRefundCardPaymentIdempotentlyPrevious(t *testing.T) {
 
 	idempotencyKey := uuid.New().String()
 
-	response1, err := PreviousApi().Payments.RefundPayment(paymentResponse.Id, &refundRequest, &idempotencyKey)
-	assert.Nil(t, err)
-	assert.NotNil(t, response1)
-
-	response2, err := PreviousApi().Payments.RefundPayment(paymentResponse.Id, &refundRequest, &idempotencyKey)
-	assert.Nil(t, err)
-	assert.NotNil(t, response2)
+	cases := []struct {
+		name           string
+		paymentId      string
+		refundRequest  payments.RefundRequest
+		idempotencyKey string
+		checker        func(*payments.RefundResponse, error)
+	}{
+		{
+			name:           "when get a refund payment with idempotencyKey request then return a response",
+			paymentId:      paymentResponse.Id,
+			refundRequest:  refundRequest,
+			idempotencyKey: idempotencyKey,
+			checker: func(response *payments.RefundResponse, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, response)
+
+				response2, err := PreviousApi().Payments.RefundPayment(paymentResponse.Id, &refundRequest, &idempotencyKey)
```

Comment:
Same as in captures. This should be another test case if it is necessary

Reply:
We need to check `assert.Equal(t, response.ActionId, response2.ActionId)`